sisl/viz/plotly/plots/pdos.py

```diff
@@ -313,24 +326,33 @@ def _after_read(self):
         """
         from xarray import DataArray
 
-        # Normalize self.PDOS to do the same treatment for both spin-polarized and spinless simulations
-        self.spin_polarized = len(self.PDOS.shape) == 3
+        # Check if the PDOS contains spin resolution (there should be three dimensions,
+        # and the first one should be the spin components)
+        self.spin = sisl.Spin.UNPOLARIZED
+        if self.PDOS.squeeze().ndim == 3:
+            self.spin = {
+                2: sisl.Spin.POLARIZED,
+                4: sisl.Spin.NONCOLINEAR
```

Comment:
and spin-orbit

Reply:
Soif I understand correctly, this implementation also makes sense for spin-orbit and there is no way to distinguish between spin-orbit and non-colinear from the shape of the PDOS array?

Reply:
exactly :)

Reply:
Then I don't know what to do here, or if I should do anything :sweat_smile:

Reply:
Are true. Here it is ok.